packages/web3-bzz/src/index.js

```diff
@@ -20,62 +20,60 @@
  * @date 2017
  */
 
-"use strict";
+import _ from 'lodash';
+import swarm from 'swarm-js';
 
-var _ = require('underscore');
-var swarm = require("swarm-js");
+let givenProvider = null;
+if (typeof ethereumProvider !== 'undefined' && ethereumProvider.bzz) {
+  givenProvider = ethereumProvider.bzz;
+}
 
+export default class Bzz {
+  static givenProvider = givenProvider
 
-var Bzz = function Bzz(provider) {
+  givenProvider = null
+  provider = null
+  pick = null
 
+  constructor (p) {
     this.givenProvider = Bzz.givenProvider;
 
+    let provider = p;
     if (provider && provider._requestManager) {
-        provider = provider.currentProvider;
+      provider = provider.currentProvider;
     }
 
     // only allow file picker when in browser
-    if(typeof document !== 'undefined') {
-        this.pick = swarm.pick;
+    if (typeof document !== 'undefined') {
+      this.pick = swarm.pick;
     }
 
     this.setProvider(provider);
-};
+  }
 
-// set default ethereum provider
-/* jshint ignore:start */
-Bzz.givenProvider = null;
```

Comment:
The point of this is to detect an existing `window/global.ethereumProvider`
You seem to have removed that altogether...

Reply:
This line here: https://github.com/jdkanani/web3.js/blob/504541d414d8b5f01d4d2d942920a5ea5c6a6256/packages/web3-bzz/src/index.js#L28 will take care of this.